fix: cursor jumping at the beginning

davidediak opened the pull request:
https://github.com/tinymce/tinymce-angular/issues/321

davidediak commented:
anybody in here? @TheSpyder @spocke

danoaky-tiny reviewed (approved):
It'd be nice to have a test for this in `FormControlTest.ts`. Thanks for the fix!

davidediak commented:
Sure, I fixed your tests and also added a test related to my fix.
Please feel free to contribute to the tests if needed, I'm not really familiar with your internal testing lib.

danoaky-tiny commented:
Ah thank you. #367 also has a fix for this failing test. Once that's merged I'll rereview

constanzaccg commented:
Hi! Is there any plan to incorporate this fix soon? Thank you

danoaky-tiny commented:
@davidediak This needs a merge from main. Just to do with the tests, won't need much change from you.

davidediak commented:
@danoaky-tiny sure, done.

EkimChau commented:
Hello [davidediak](https://github.com/davidediak),

Thank you for your code contribution, in order for Tiny Inc to accept your code, we will need permission to use your code contribution, this can be done through the Tiny Contributor License Agreement (Tiny CLA). Please contact [legal@tiny.cloud](mailto:legal@tiny.cloud) to sign the Tiny CLA agreement.

davidediak commented:
@EkimChau you have my permission to use my code, given the fact that I spent time fixing the bug and did open a PR.
I have no intention of proving your company with my personal data.

TheSpyder commented:
This is clearly stated in the contributing guide. We don't have an automated system for it at the moment.
https://github.com/tinymce/tinymce-angular/blob/5bcfde80e8c5cb78b806d14c5c9e11d1af2c3471/CONTRIBUTING.md?plain=1#L8

The CLA protects us, our open-source community and our customers from the distribution of unauthorised code. It's non-negotiable, sorry.

carlosmintfan commented:
@davidediak I‘ll use your PR if it looks good to me once I fork the Angular integration for my [HugeRTE](https://hugerte.org) fork of TinyMCE.

joe-baudisch commented on tinymce-angular-component/src/main/ts/editor/editor.component.ts:
const cursor = this._editor.selection.getBookmark(3) as any
    if (cursor['start'] && cursor['start'].startsWith('text()[0]')) {
      // do nothing because the cursor initially jumps back to first line with 'text()[0]'
      // while pasting in text from outside; moving to this bookmark after set content
      // creates this cursor jump which should be avoided;
      // if you paste in text from inside the editor or write text by keyboard
      // the cursor starts with a paragraph tag created by TMCE like
      // 'p[element_index]/text()[0]'
    } else {
      try {
        this._editor.selection.moveToBookmark(cursor);
      } catch (e) {
        /* ignore */
      }
    }
